src/document_to_podcast/cli.py

```diff
@@ -0,0 +1,153 @@
+import re
+from pathlib import Path
+
+import numpy as np
+import soundfile as sf
+import yaml
+from fire import Fire
+from loguru import logger
+
+
+from document_to_podcast.config import Config, Speaker, DEFAULT_PROMPT, DEFAULT_SPEAKERS
+from document_to_podcast.inference.model_loaders import (
+    load_llama_cpp_model,
+    load_parler_tts_model_and_tokenizer,
+)
+from document_to_podcast.inference.text_to_text import text_to_text_stream
+from document_to_podcast.inference.text_to_speech import text_to_speech
+from document_to_podcast.preprocessing import DATA_CLEANERS, DATA_LOADERS
+
+
+@logger.catch(reraise=True)
+def document_to_podcast(
+    input_file: str | None = None,
+    output_folder: str | None = None,
+    text_to_text_model: str = "allenai/OLMoE-1B-7B-0924-Instruct-GGUF/olmoe-1b-7b-0924-instruct-q8_0.gguf",
+    text_to_text_prompt: str = DEFAULT_PROMPT,
+    text_to_speech_model: str = "parler-tts/parler-tts-mini-v1",
+    speakers: list[Speaker] | None = None,
+    from_config: str | None = None,
+):
+    """
+    Generate a podcast from a document.
+
+    Args:
+        input_file (str): The path to the input file.
+            Supported extensions:
+
+                - .pdf
+                - .html
+                - .txt
+                - .docx
+                - .md
+
+        output_folder (str): The path to the output folder.
+            Two files will be created:
+
+                - {output_folder}/podcast.txt
+                - {output_folder}/podcast.wav
+
+        text_to_text_model (str, optional): The path to the text-to-text model.
+
+            Need to be formatted as `owner/repo/file`.
+
+            Need to be a gguf file.
+
+            Defaults to `allenai/OLMoE-1B-7B-0924-Instruct-GGUF/olmoe-1b-7b-0924-instruct-q8_0.gguf`.
+
+        text_to_text_prompt (str, optional): The prompt for the text-to-text model.
+            Defaults to DEFAULT_PROMPT.
+
+        text_to_speech_model (str, optional): The path to the text-to-speech model.
+            Defaults to `parler-tts/parler-tts-mini-v1`.
+
+        speakers (list[Speaker] | None, optional): The speakers for the podcast.
+            Defaults to DEFAULT_SPEAKERS.
+
+        from_config (str, optional): The path to the config file. Defaults to None.
+
+            If provided, all other arguments will be ignored.
+    """
+    if from_config:
+        config = Config.model_validate(yaml.safe_load(Path(from_config).read_text()))
+    else:
+        speakers = speakers or DEFAULT_SPEAKERS
+        config = Config(
+            input_file=input_file,
+            output_folder=output_folder,
+            text_to_text_model=text_to_text_model,
+            text_to_text_prompt=text_to_text_prompt,
+            text_to_speech_model=text_to_speech_model,
```

Comment:
There is a tiny type hint issue here. In line 27 of this file, we type it as `str` but in the Config its a Literal of our supported models.

One way to resolve this would be to type `text_to_speech_model` in Config as `str` and then have a `validate_text_to_speech_model` to make sure a supported model is selected. 

Another way is to type it in the `document_to_podcast` as Literal and we define somewhere a `SUPPORTED_TTS_MODELS = [...]` list that we reference it here and in the `config.py`

The first approach follows the logic of the other validators you have defined (`validate_text_to_text_model`), but on the other hand it is also a bit more limiting. 

Like what if the user puts a different model that we dont have in our "supported models" list but they get lucky (maybe a newer version of Oute/Parler models) and it works? Also in the second approach we should put a comment like "these are verified models, but maybe other models might work"

Reply:
I added a SUPPORTED_TTS_MODELS literal to the config file and imported it in cli.py.

> Like what if the user puts a different model that we dont have in our "supported models" list but they get lucky (maybe a newer version of Oute/Parler models) and it works? 

I think using an arbitrary model is supported already by using the API directly, we can document it in the customization guide. 
If there is a new version of one of the models we support, we can just extend the list ourselves

Reply:
Sounds good. But yeah, after #49 , we need to go through the docs again and add some notes.